javascript-engine/src/strategies.ts

```diff
@@ -0,0 +1,95 @@
+import type { Context, State, StrategyDefinition } from './types'
```

Comment:
Can I ask for some tests here? This is likely going to break in the future when this needs to handle deltas and it would be super cool if a random person got warned of that

Reply:
Addressed the delta handling here: https://github.com/Unleash/yggdrasil/pull/226/commits/df93d7dc4911d570ef04ec299c1e2d298b3e55f4